<?xml version="1.0" ?>
<robot name="primary_manipulator" xmlns:xacro="http://www.ros.org/wiki/xacro">

<xacro:include filename="$(find cooking_robot)/urdf/materials.xacro" />
<xacro:include filename="$(find cooking_robot)/urdf/primary_manipulator.trans" />
<xacro:include filename="$(find cooking_robot)/urdf/primary_manipulator.gazebo" />
<link name="world">
  
</link>
<link name="base_link">
  <inertial>
    <origin xyz="3.957301377503479e-05 5.550169171063192e-09 0.0412767136677171" rpy="0 0 0"/>
    <mass value="2.066513090661844"/>
    <inertia ixx="0.006011" iyy="0.006018" izz="0.009572" ixy="-0.0" iyz="-0.0" ixz="-2e-06"/>
  </inertial>
  <visual>
    <origin xyz="0 0 0" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://cooking_robot/meshes/base_link.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin xyz="0 0 0" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://cooking_robot/meshes/base_link.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>
<joint name="j0" type="fixed">
  <parent link="world"/>
  <child link="base_link"/>
  <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
</joint>


<link name="link1_1">
  <inertial>
    <origin xyz="0.0230637786360918 0.001341583374922287 0.046898108796872714" rpy="0 0 0"/>
    <mass value="5.742113360427433"/>
    <inertia ixx="0.009895" iyy="0.017173" izz="0.013403" ixy="-0.000482" iyz="-0.000382" ixz="-0.005786"/>
  </inertial>
  <visual>
    <origin xyz="-0.0 -0.0 -0.07" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://cooking_robot/meshes/link1_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin xyz="-0.0 -0.0 -0.07" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://cooking_robot/meshes/link1_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<link name="link2_1">
  <inertial>
    <origin xyz="0.08046108742370087 0.03265671734962108 -2.2244900366175102e-11" rpy="0 0 0"/>
    <mass value="4.349837143782598"/>
    <inertia ixx="0.002088" iyy="0.025024" izz="0.024204" ixy="0.000167" iyz="0.0" ixz="-0.0"/>
  </inertial>
  <visual>
    <origin xyz="-0.09 -0.015 -0.17" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://cooking_robot/meshes/link2_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin xyz="-0.09 -0.015 -0.17" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://cooking_robot/meshes/link2_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<link name="link3-2_joint_1">
  <inertial>
    <origin xyz="-5.774756728360586e-08 -0.04520965854239076 0.0029520539853202243" rpy="0 0 0"/>
    <mass value="0.48926159770428396"/>
    <inertia ixx="0.0005" iyy="0.000169" izz="0.000447" ixy="0.0" iyz="2e-06" ixz="-0.0"/>
  </inertial>
  <visual>
    <origin xyz="-0.29 -0.034 -0.17" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://cooking_robot/meshes/link3-2_joint_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin xyz="-0.29 -0.034 -0.17" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://cooking_robot/meshes/link3-2_joint_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<link name="link3_1">
  <inertial>
    <origin xyz="2.4722203617599803e-08 -0.0005180384055503261 0.06152943396730581" rpy="0 0 0"/>
    <mass value="1.241116625279245"/>
    <inertia ixx="0.002432" iyy="0.001722" izz="0.001103" ixy="0.0" iyz="3e-05" ixz="-0.0"/>
  </inertial>
  <visual>
    <origin xyz="-0.29 0.0135 -0.195" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://cooking_robot/meshes/link3_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin xyz="-0.29 0.0135 -0.195" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://cooking_robot/meshes/link3_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<link name="gripper_joint1_1">
  <inertial>
    <origin xyz="-2.1296181484942167e-10 0.04520965864214052 0.002952063965921292" rpy="0 0 0"/>
    <mass value="0.4892591904426007"/>
    <inertia ixx="0.0005" iyy="0.000169" izz="0.000447" ixy="0.0" iyz="-2e-06" ixz="0.0"/>
  </inertial>
  <visual>
    <origin xyz="-0.29 0.0595 -0.3125" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://cooking_robot/meshes/gripper_joint1_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin xyz="-0.29 0.0595 -0.3125" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://cooking_robot/meshes/gripper_joint1_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<link name="link4_1">
  <inertial>
    <origin xyz="0.00018388764296389803 0.00027077966008015597 0.025509584507645178" rpy="0 0 0"/>
    <mass value="0.6801945828664837"/>
    <inertia ixx="0.000469" iyy="0.000457" izz="0.000272" ixy="0.0" iyz="0.0" ixz="0.0"/>
  </inertial>
  <visual>
    <origin xyz="-0.29 0.012 -0.3375" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://cooking_robot/meshes/link4_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin xyz="-0.29 0.012 -0.3375" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://cooking_robot/meshes/link4_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<link name="Spur_Gear__24_teeth___1__1">
  <inertial>
    <origin xyz="-0.010000185471326528 -0.0027254969628815826 0.030042323969180884" rpy="0 0 0"/>
    <mass value="0.19629204650966112"/>
    <inertia ixx="0.000182" iyy="0.000182" izz="1.4e-05" ixy="0.0" iyz="1.9e-05" ixz="0.0"/>
  </inertial>
  <visual>
    <origin xyz="-0.30018 -0.001765 -0.4125" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://cooking_robot/meshes/Spur_Gear__24_teeth___1__1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin xyz="-0.30018 -0.001765 -0.4125" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://cooking_robot/meshes/Spur_Gear__24_teeth___1__1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<link name="Spur_Gear__24_teeth__1">
  <inertial>
    <origin xyz="-0.010000185471326528 0.0027270129816117927 0.03010843991676121" rpy="0 0 0"/>
    <mass value="0.19647451990768114"/>
    <inertia ixx="0.000183" iyy="0.000182" izz="1.4e-05" ixy="0.0" iyz="-1.9e-05" ixz="0.0"/>
  </inertial>
  <visual>
    <origin xyz="-0.30018 0.025235 -0.4125" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://cooking_robot/meshes/Spur_Gear__24_teeth__1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin xyz="-0.30018 0.025235 -0.4125" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://cooking_robot/meshes/Spur_Gear__24_teeth__1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<joint name="joint_base" type="revolute">
  <origin xyz="0.0 0.0 0.07" rpy="0 0 0"/>
  <parent link="base_link"/>
  <child link="link1_1"/>
  <axis xyz="0.0 0.0 -1.0"/>
  <limit lower="-1.571" upper="1.571" effort="100" velocity="100"/>
  
</joint>

<joint name="joint_link_1" type="revolute">
  <origin xyz="0.09 0.015 0.1" rpy="0 0 0"/>
  <parent link="link1_1"/>
  <child link="link2_1"/>
  <axis xyz="-0.0 1.0 0.0"/>
  <limit lower="-2.42" upper="0.71" effort="100" velocity="100"/>
</joint>

<joint name="joint_link_2" type="revolute">
  <origin xyz="0.2 0.019 0.0" rpy="0 0 0"/>
  <parent link="link2_1"/>
  <child link="link3-2_joint_1"/>
  <axis xyz="0.0 1.0 0.0"/>
  <limit lower="-1.0" upper="3.14" effort="100" velocity="100"/>
</joint>

<joint name="joint_link_3" type="revolute">
  <origin xyz="0.0 -0.0475 0.025" rpy="0 0 0"/>
  <parent link="link3-2_joint_1"/>
  <child link="link3_1"/>
  <axis xyz="0.0 0.0 -1.0"/>
  <limit lower="-1.571" upper="1.571" effort="100" velocity="100"/>
</joint>

<joint name="joint_link_4" type="revolute">
  <origin xyz="0.0 -0.046 0.1175" rpy="0 0 0"/>
  <parent link="link3_1"/>
  <child link="gripper_joint1_1"/>
  <axis xyz="-0.0 -1.0 0.0"/>
  <limit lower="-1.571" upper="1.571" effort="100" velocity="100"/>
</joint>

<joint name="joint_gripper" type="revolute">
  <origin xyz="0.0 0.0475 0.025" rpy="0 0 0"/>
  <parent link="gripper_joint1_1"/>
  <child link="link4_1"/>
  <axis xyz="0.0 0.0 -1.0"/>
  <limit lower="-1.571" upper="1.571" effort="100" velocity="100"/>
</joint>

<joint name="joint_gripper_left" type="revolute">
  <origin xyz="0.01018 0.013765 0.075" rpy="0 0 0"/>
  <parent link="link4_1"/>
  <child link="Spur_Gear__24_teeth___1__1"/>
  <axis xyz="1.0 -0.0 0.0"/>
  <limit lower="-0.68" upper="0" effort="100" velocity="100"/>
</joint>

<joint name="joint_gripper_right" type="revolute">
  <origin xyz="0.01018 -0.013235 0.075" rpy="0 0 0"/>
  <parent link="link4_1"/>
  <child link="Spur_Gear__24_teeth__1"/>
  <axis xyz="1.0 -0.0 0.0"/>
  <limit lower="0" upper="0.68" effort="100" velocity="100"/>
</joint>

</robot>
